test/api/test_deskew_api.py

```diff
@@ -0,0 +1,161 @@
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#   FROM,  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#   DEALINGS IN THE SOFTWARE.
+#  </summary>
+#  ----------------------------------------------------------------------------
+
+from itertools import product
+
+import asposeimagingcloud.models.requests as requests
+from test.api.imaging_api_tester import ImagingApiTester
+
+
+class TestDeskewApi(ImagingApiTester):
+    """ Class for testing DeskewAPI """
+
+    def test_deskew_image(self):
+        """ Test deskew_image """
+
+        if not self.EXTENDED_TEST:
+            format_extension_test_cases = ['.jpg']
+        else:
+            format_extension_test_cases = [
+                '.jpg',
+                '.bmp',
+                #'.dicom',
+                '.gif',
+                '.j2k',
+                '.png',
+                '.psd',
+                '.tiff',
+                '.webp'
+                #'.dng',
+                #'.djvu'
+                ]
+
+        for format_extension in format_extension_test_cases:
+            with self.subTest('format_extension: ' + str(format_extension)):
+                folder = self.temp_folder
+                storage = self.test_storage
+                resize_proportinoally = True
+                bk_color = "green"
+
+                def request_invoker():
+                    return self.imaging_api.deskew_image(
+                        requests.DeskewImageRequest(name, resize_proportinoally, bk_color, folder, storage))
+
+                def properties_tester(
+                        original_properties,
+                        result_properties,
+                        result_stream):
+                    self.assertIsNotNone(result_stream)
+
+                for input_file in self.input_test_files:
+                    if not str(input_file.name).endswith(format_extension):
+                        continue
+
+                    name = input_file.name
+
+                    self.get_request_tester(
+                        'DeskewImageTest',
+                        'Input image: {0}; Output format: {1}; Resize proportionally: {2}; BkColor: {3};'.format(
+                            name,
+                            format_extension,
+                            resize_proportinoally,
+                            bk_color
+                        ),
+                        name,
+                        request_invoker,
+                        properties_tester,
+                        folder,
+                        storage)
+
+    def test_create_deskewed_image(self):
+        """ Test create_deskewed_image"""
+
+        if not self.EXTENDED_TEST:
+            format_extension_test_cases = ['.jpg']
+        else:
+            format_extension_test_cases = [
+                '.jpg',
+                '.bmp',
+                #'.dicom',
```

Comment:
@dmitrypasyutin Please, provide a comment with the reason why some of the test cases are commented out, so we could enable them when the appropriate feature is implemented on the server side, for example.